sources/packages/api/src/route-controllers/restriction/restriction.controller.ts

```diff
@@ -35,10 +67,192 @@ export class RestrictionController extends BaseController {
+   * @returns Categories option list.
+   */
+  @Groups(UserGroups.AESTUser)
+  @AllowAuthorizedParty(AuthorizedParties.aest)
+  @Get("/categories/options-list")
+  async getCategoriesOptionsList(): Promise<OptionItem[]> {
+    const categories =
+      await this.restrictionService.getAllRestrictionCategories();
+    return categories.map((category) => ({
+      id: category.id,
+      description: category.restrictionCategory,
+    }));
+  }
+
+  /**
+   * REST API to provide the list of restriction reasons for selected category.
+   * @returns Reasons option list.
+   */
+  @Groups(UserGroups.AESTUser)
+  @AllowAuthorizedParty(AuthorizedParties.aest)
+  @Get("/reasons/options-list/category/:restrictionCategory")
+  async getReasonsOptionsList(
+    @Param("restrictionCategory") restrictionCategory: string,
+  ): Promise<OptionItem[]> {
+    const reasons =
+      await this.restrictionService.getRestrictionReasonsByCategory(
+        restrictionCategory,
+      );
+    return reasons.map((reason) => ({
+      id: reason.id,
+      description: `${reason.restrictionCode} - ${reason.description}`,
+    }));
+  }
+
+  /**
+   * Rest API to get the details for view student restriction.
+   * @param studentId
+   * @param restrictionId
+   * @returns Student restriction detail view.
+   */
+  @Groups(UserGroups.AESTUser)
+  @AllowAuthorizedParty(AuthorizedParties.aest)
+  @Get("/student/:studentId/studentRestriction/:studentRestrictionId")
+  async getStudentRestrictionDetail(
+    @Param("studentId") studentId: number,
+    @Param("studentRestrictionId") studentRestrictionId: number,
+  ): Promise<StudentRestrictionDetail> {
+    const studentRestriction =
+      await this.studentRestrictionService.getStudentRestrictionDetailsById(
+        studentId,
+        studentRestrictionId,
+      );
+    if (!studentRestriction) {
+      throw new NotFoundException("The student restriction does not exist.");
+    }
+    return {
+      restrictionId: studentRestriction.id,
+      restrictionType: studentRestriction.restriction.restrictionType,
+      restrictionCategory: studentRestriction.restriction.restrictionCategory,
+      description: studentRestriction.restriction.description,
+      createdAt: studentRestriction.createdAt,
+      updatedAt: studentRestriction.updatedAt,
+      createdBy: studentRestriction.creator
+        ? `${studentRestriction.creator.lastName}, ${studentRestriction.creator.firstName}`
+        : "",
+      updatedBy: studentRestriction.modifier
+        ? `${studentRestriction.modifier.lastName}, ${studentRestriction.modifier.firstName}`
+        : "",
+      isActive: studentRestriction.isActive,
+      restrictionNote: studentRestriction.restrictionNote?.description,
+      resolutionNote: studentRestriction.resolutionNote?.description,
+    };
+  }
+
+  /**
+   * Rest API to add a new provincial restriction to Student.
+   * Note: Federal restrictions are added/resolved by nightly job not through this API.
```

Comment:
you can give `* *` for important comments, (https://marketplace.visualstudio.com/items?itemName=aaron-bond.better-comments)

Reply:
Addressed.